src/pages/StocktakeEditPage.js

```diff
@@ -120,7 +120,7 @@ export class StocktakeEditPage extends React.Component {
     const { database } = this.props;
     const queryString = 'type == $0 && isActive == true';
     const reasons = database.objects('Options').filtered(queryString, 'stocktakeLineAdjustment');
-    this.setState({ usesReasons: reasons.length !== 0 });
+    this.setState({ reasons });
```

Comment:
Much better. Wondering how you feel about having just a constant in the constructor: `this.USES_REASONS = null/false` and assigning `!!reasons.length` in CDM - saves a bit of fluffing around throughout the component. As it would be static it seems right to have it outside of state, even though I don't really like it!

Reply:
I think it's fine as is